index.js

```diff
@@ -743,6 +743,29 @@ exports.encodingLength = function (result) {
     encodingLengthList(result.additionals || [], answer)
 }
 
+exports.streamEncode = function (result) {
+  const len = exports.encodingLength(result)
+  const buf = Buffer.allocUnsafe(len + 2)
+  exports.encode(result, buf, 2)
+  buf.writeUInt16BE(len, 0)
+  exports.streamEncode.bytes = len + 2
+  return buf
+}
+
+exports.streamEncode.bytes = 0
+
+exports.streamDecode = function (buf) {
+  const len = buf.readUInt16BE(0)
+  if (buf.length < len) {
+    return {}
```

Comment:
in case we end up wanting to land this, this should return `null` or throw in this case